tests/register/register.go

```diff
@@ -31,6 +38,68 @@ func init() {
 	Tests = make(map[TestType][]types.Test)
 }
 
+var count = 0
+
 func Register(tType TestType, t types.Test) {
 	Tests[tType] = append(Tests[tType], t)
+	// count += howmany(t.Config)
+	count++
+	fmt.Printf("count now at: %d\n", count)
+}
+
+// Finds version, creates Test structs for all compatible versions and registers them
+func RegisterTest(tType TestType, t types.Test) {
+	versions := [][]string{
+		{"0.0.0"},
+		{"1.0.0"},
+		{"2.0.0", "2.1.0", "2.2.0", "2.3.0-experimental"},
+	}
+
+	pattern := regexp.MustCompile("[0-9]{1}.[0-9]{1}.[0-9]{1}")
+	versionArr := pattern.FindAllStringSubmatch(t.Config, 1)
```

Comment:
Because you're already making an assumption that if the version is present it will always be the first submatch found you can switch this to `FindStringSubmatch` and then just deal with the first submatch found in the string.